### ISSUE TITLE
propose: create draft proposals

### ISSUE THREAD

**Issue by blaggacao** [git-town/git-town#1519]
I know at least of the following hosting drivers to support WIP PR:
- Gitlab
- Github
- Gitea

Therefore:
- `git town expose`:  alias to `git town new-pull-request` but prepend with `WIP: ` -> draft state
- `git town expose --unblock` -> removes WIP
- `git town ship`  -- blocks on WIP branches and returns an error

requires: https://github.com/git-town/git-town/pull/1518#discussion_r434970545

**Comment by kevgo** [comment]
GitHub uses a different model where draft pull requests have a dedicated status, not just "WIP: " in front of them. See https://github.blog/2019-02-14-introducing-draft-pull-requests for details.

![Screenshot 2020-06-04 at 8 07 25 AM](https://user-images.githubusercontent.com/268934/83760472-9a952100-a63a-11ea-9605-df7c74c67716.png)

![Screenshot 2020-06-04 at 8 09 06 AM](https://user-images.githubusercontent.com/268934/83760637-d4662780-a63a-11ea-8f94-c63c112c325e.png)

What about these commands:

- **git town new-pull-request** creates a normal pull request
- **git town draft-pull-request** creates a draft pull request
- **git town finish-pull-request** removes the draft status of the PR

The CodeHostingDriver instances implement how draft pull requests are done on the respective platform. On GitHub, they set the PR status. On other platforms, they might add or remove `WIP: ` to the pull request title.

Alternatively, if we don't want to create so many new commands:

- **git town pull-request** creates a normal pull request
- **git town pull-request new** creates a normal pull request
- **git town pull-request draft** creates a draft pull request
- **git town pull-request finish** removes the draft status of the PR for the current branch

This is a breaking change of the UI. But I'm okay with that since it is a lot cleaner and keeps the number of top-level commands smaller.

**Comment by blaggacao** [comment]
Alternative C looks best, indeed.

**Comment by kevgo** [comment]
Okay, let's go with alternative C then!

**Comment by kevgo** [comment]
I think this is out of scope. Creating proposals and changing their status is supposed to be done via the web UI.

**Comment by ruudk** [comment]
Could we reconsider? Today I proposed a PR that I wanted to be a draft, but it immediately notified code owners for review. I then quickly switched it back to Draft, but the notifications were already sent out and people started reviewing.

**Comment by kevgo** [comment]
How did you propose your PR?

If you use the GitHub API connector (the one that uses tokens), then you create the PR yourself using the GitHub UI. Just create the PR as a draft there. 

If you use the `gh` connector, we run `gh pr create`, and `gh` will offer you an option to create a draft PR.

So all the options are there.

Assuming we would implement a `git town propose --draft` flag, we could call `gh pr create --draft`, and that would work. But there is no option to create draft PRs using the GitHub API connector. See [the available URL parameters](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/using-query-parameters-to-create-a-pull-request). I'm a bit hesitant to add an option that can't be supported consistently across most forge connectors.

Maybe we show a warning if the user has enabled draft proposals via a hypothetical `git-town.propose-drafts` config option, and the configured connector doesn't allow creating draft proposals?

**Comment by kevgo** [comment]
After thinking about this more, I have created https://github.com/git-town/git-town/issues/6079 to track the idea of a `--draft` config option. Thanks for the inspiration!
